File: solutions/0201. Bitwise AND of Numbers Range/0201.js
// Time complexity: O(1)
// Space complexity: O(1)
var rangeBitwiseAnd = function(left, right) {
  let shiftBits = 0;

  while (left != right) {
    left >>= 1;
    right >>= 1;
    ++shiftBits;
  }

  return left << shiftBits;
};

File: solutions/0201. Bitwise AND of Numbers Range/0201.py
# Time complexity: O(1)
# Space complexity: O(1)
class Solution:
  def rangeBitwiseAnd(self, left: int, right: int) -> int:
    shiftBits = 0

    while left != right:
      left >>= 1
      right >>= 1
      shiftBits += 1

    return left << shiftBits 